### PR TITLE
Single store mode incorrect scope values in plp fix

### LINKED ISSUE #25933: Single Store Mode- Incorrect scope values using at Storefront for Catalog Section
### Preconditions (*)

1. Magento 2.3.3 CE & 2.4-develop with sample data
2. Magento Instance should have one **Website**, **Store** and **Store View** only.

### Steps to reproduce (*)

1. Under Default Config, Navigate to Stores > Settings > Configuration > Catalog > Catalog
2. Change the values of field Products per Page on Grid Allowed Values from **9,15,30** to **5,10,20,40,80** (Uncheck **Use System value** checkbox if ticked - optional)
3. Change the values of field Products per Page on Grid Default Value from **9** to **10**  (Uncheck **Use System value** checkbox if ticked- optional)
4. Click on **Save Config**
5. Clear the cache if prompted to refresh the cache
6. Switch the scope from **Default Config** to **Default Store View** Scope from the drop-down showing on upper-left corner
7. Navigate to Stores > Settings > Configuration > Catalog > Catalog
8. From Default Store View Scope, Uncheck **Use Website** checkbox for following fields
- Products per Page on Grid Allowed Values
- Products per Page on Grid Default Value
9.  Change the values of field Products per Page on Grid Allowed Values to 30,60,90
10. Change the values of field Products per Page on Grid Default Value	to 30
11. Click on **Save Config**
11. Switch the scope to **Default Config** from **Default Store View Scope**
12. From Default Config Scope, Set Enable Single-Store Mode to **Yes**
13. Click **Save Config**
14. Clear all the cache
15. Now, Navigate to Stores > Settings > Configuration > Catalog > Catalog
16. Tries to update the values for following fields.
- Products per Page on Grid Allowed Values **20,40,60**
- Products per Page on Grid Default Value **20**
17. **Save Config** and Clear all the cache
18. Open any category page to see the **Show per page** values.

### Expected result (*)

1. On category page, it should show 20,40,60 in the **Show per page** drop-down.

### Actual result (*)

1. On category page, it shows values as 30,60,90  in the **Show per page** drop-down.

### Additional Info
1. The issue may be reproduced for all other fields which are available on `showInStore=1`

### PR DESCRIPTION
**Pull request by muthu-ziffity**
### Description

In the category page, In single store mode Show per page section doesn't get the value from admin properly. Now it got fixed in this Pull Request

### Fixed Issue

1. Fixes magento/magento2#25933

### Reference for the fix

pageView

### CODE REVIEW

File: lib/internal/Magento/Framework/App/Config.php [hunk 1]
```diff
@@ -12,7 +12,7 @@
 use Magento\Framework\App\Config\ScopeConfigInterface;
 
 /**
- * Class Config
+ * Class Config for get value of fields
  */
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Config.php:16]: Just remove that redundant annotation

File: lib/internal/Magento/Framework/App/Config.php [hunk 2]
```diff
@@ -21,6 +21,11 @@ class Config implements ScopeConfigInterface
      */
     const CACHE_TAG = 'CONFIG';
 
+    /**
+     * Config single store mode tag
+     */
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Config.php:26]: `const` does not require annotations. This annotation does not bring any value.

File: lib/internal/Magento/Framework/App/Config.php [hunk 3]
```diff
@@ -31,6 +36,8 @@ class Config implements ScopeConfigInterface
      */
     private $types;
 
+    public $isSingleStoreModeEnabled;
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Config.php:39]: 1. That should not be public property (change to private)
2. Add PHPDoc for the property

File: lib/internal/Magento/Framework/App/Config.php [hunk 4]
```diff
@@ -80,6 +90,26 @@ public function getValue(
         return $this->get('system', $configPath);
     }
 
+    /**
+     * Check single store mode in the system
+     *
+     * @param string $scope
+     * @return mixed
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Config.php:97]: Return type of all `is...` methods should be `bool`

File: lib/internal/Magento/Framework/App/Config.php [hunk 5]
```diff
@@ -80,6 +90,26 @@ public function getValue(
...
+        /** caching the single store mode field value in variable for avoid multiple database hit */
+        if (!isset($this->isSingleStoreModeEnabled)) {
+            $this->isSingleStoreModeEnabled = (bool)$this->get('system', self::SINGLE_STORE_MODE_TAG);
+        }
+
+        /** set scope based on single store mode field value */
+        if (!empty($this->isSingleStoreModeEnabled)) {
+            $scope = ScopeConfigInterface::SCOPE_TYPE_DEFAULT;
+        }
+        return $scope;
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Config.php:110]: The method does not return information whether it's `Single Store Mode` or not.

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Thank you for your contribution. Please review the comments.

**Comment by muthu-ziffity** [comment]
@lbajsarowicz  Thanks for reviewing my PR. I will update those mentioned changes and let you know to check again

**Comment by engcom-Charlie** [comment]
Hi @muthu-ziffity, I'm closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for your collaboration.

**Pull request closed without merging**